usecases/auth/authorization/rbac/manager.go

```diff
@@ -41,6 +42,11 @@ func New(rbacStoragePath string, rbac rbacconf.Config, logger logrus.FieldLogger
 
 func (m *manager) UpsertRolesPermissions(roles map[string][]authorization.Policy) error {
 	for roleName, policies := range roles {
+		// assign role to internal user to make sure to catch empty roles
+		// e.g. : g, user:wv_internal_empty, role:roleName
+		if _, err := m.casbin.AddRoleForUser(conv.PrefixUserName(conv.InternalPlaceHolder), conv.PrefixRoleName(roleName)); err != nil {
+			return err
+		}
```

Comment:
**Question (non-blocking):** Does this mean that every user-created role will be assigned to the internal user `conv.InternalPlaceHolder`? Do you think this might have a significant performance impact at scale?

Reply:
nope, this means every role will have user to be a place holder for empty and stale roles

Reply:
this is for role not permissions , so it will be only 1 for role no matter how many permissions got added